api/payloads/service_binding.go

```diff
@@ -16,18 +18,40 @@ type ServiceBindingCreate struct {
 }
 
 func (p ServiceBindingCreate) ToMessage(spaceGUID string) repositories.CreateServiceBindingMessage {
+	var appGUID string
+	if p.Relationships.App != nil {
+		appGUID = p.Relationships.App.Data.GUID
+	}
+
 	return repositories.CreateServiceBindingMessage{
 		Name:                p.Name,
 		ServiceInstanceGUID: p.Relationships.ServiceInstance.Data.GUID,
-		AppGUID:             p.Relationships.App.Data.GUID,
+		AppGUID:             appGUID,
 		SpaceGUID:           spaceGUID,
+		Type:                p.Type,
 	}
 }
 
 func (p ServiceBindingCreate) Validate() error {
 	return jellidation.ValidateStruct(&p,
-		jellidation.Field(&p.Type, validation.OneOf("app")),
-		jellidation.Field(&p.Relationships, jellidation.NotNil),
+		jellidation.Field(&p.Type, validation.OneOf("app", "key")),
+		jellidation.Field(&p.Relationships, jellidation.By(func(value any) error {
```

Comment:
This validation rule can be simplified a bit:

```
func (p ServiceBindingCreate) Validate() error {
	return jellidation.ValidateStruct(&p,
		jellidation.Field(&p.Type, validation.OneOf("app", "key")),
		jellidation.Field(&p.Relationships, jellidation.Required),
		jellidation.Field(&p.Relationships.App, jellidation.Required.When(p.Type == "app")),
		jellidation.Field(&p.Relationships.App.Data.GUID, jellidation.Required.When(p.Type == "app")),
	)
}
```

We can make use of the `jellidation.NotNil` validation rule. I suspect it can be simplified even more but couldn't... It's a complicated conditional validation anyway.